avocado/core/nrunner_avocado_instrumented.py

```diff
@@ -51,7 +51,11 @@ def _run_avocado(runnable, queue):
         state = instance.get_state()
         # This should probably be done in a translator
         if 'status' in state:
-            state['status'] = state['status'].lower()
+            status = state['status'].lower()
+            if status in ['pass', 'fail', 'skip', 'error']:
+                state['result'] = status
+            state['status'] = status
```

Comment:
I might be wrong, but here we are bringing back the "ambiguity" right?

Result = [pass|fail|skip|error] = the test result.
Status = [started|running|finished|aborted] = the runner status.

IIUC here, we are adding to status the same status as in result. Not sure. Makes sense? Or did I understood it wrong?

Reply:
You're right. The `avocado-instrumented` support is a bit more complex because of the amount the amount of key/values the underlying `INSTRUMENTED` test produces, something like:

```python
{'logdir': '/tmp/tmpejlwqloe/test-results/1-PassTest.test', 'logfile': '/tmp/tmpejlwqloe/test-results/1-PassTest.test/debug.log', 'status': 'pass', 'running': False, 'paused': False, 'time_elapsed': 0.0004825592041015625, 'time_end': 1587045902.0333574, 'fail_reason': None, 'fail_class': None, 'traceback': None, 'timeout': None, 'whiteboard': '', 'phase': 'FINISHED', 'class_name': 'PassTest', 'job_logdir': None, 'job_unique_id': 'cabba3850f899aa6256cf55f7ff5c605b51cbe24', 'params': [], 'result': 'pass', 'id': '1-examples/tests/passtest.py:PassTest.test'}
```

And there are some conflicting keys, such as `running = True|False`, which can conflict with the idea of `status = finished`. So in the future we'll have to revisit a lot of these.

For now, I'll be setting  the `result` key to either one of 'pass', 'fail', 'skip', 'error', and if on one of these, also set `status` to `finished` which is required by the status server from all test types.